src/site/_collections/post-to-collections.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module.exports = (collections) => {
+  const postToCollections = {};
+  const paths = collections
+    .getFilteredByTag("pathItem")
+    .map((p) => p.data.path);
+
+  paths.forEach((path) => {
+    const details = {
+      title: path.title,
+      slug: path.slug,
+      href: `/${path.slug}/`,
+    };
+
+    path.topics.forEach((topic) => {
+      topic.pathItems.forEach((pathItem) => {
```

Comment:
Also suggest using the variable name `postSlug` instead of `pathItem`. I think then you'll end up with:

```
postToCollections[postSlug][path.slug]
```

which might be easier to grok at a glance.